Add class analysis (#284)
ServiceNow/azimuth: docs/docs/key-concepts/class-similarity.md

# Class Similarity

## What is it?

> Class analysis is an aspect of similarity analysis. As such, it is based on the relative locations
> of utterances in embedding space, and only requires training data to be supplied.

Reviewer: ```suggestion
Class analysis is an aspect of similarity analysis, so it is based on the relative locations of utterances in embedding space. Because of this, it only requires training data to be supplied.
```

Reviewer: We need to sort out class similarity, class analysis, and class overlap. In the meantime, I made these sentences technically correct. (I.e., the class analysis table also includes model confusion, so it _includes_ an aspect of similarity analysis, but it is not _just_ an aspect of similarity analysis.)
